data_spec_validator/spec/validators.py

```diff
@@ -449,3 +454,36 @@ def validate(value, extra, data) -> Tuple[bool, Union[Exception, str]]:
         ok = type(value) == str and match_func and match_func(pattern, value)
         info = '' if ok else ValueError(f'{repr(value)} does not match "{error_regex_param}"')
         return ok, info
+
+
+class CondExistValidator(BaseValidator):
+    name = COND_EXIST
+
+    @staticmethod
+    def validate(value, extra, data) -> Tuple[bool, Union[Exception, str]]:
+        allow_unknown = extra.get('ALLOW_UNKNOWN', False)
+        params = extra.get(CondExistValidator.name, {})
+        any_keys = params.get('ANY', [])
+        must_with_keys = params.get('WITH', [])
+        must_without_keys = params.get('WITHOUT', [])
+
+        if isinstance(value, UnknownFieldValue) and not allow_unknown:
+            return False, LookupError('must exist')
+
+        ok = True
+        msg = ''
+        if must_with_keys and not isinstance(value, UnknownFieldValue):
+            ok = all([key in data for key in must_with_keys])
+            msg = f'{", ".join(must_with_keys)} must exist' if not ok else msg
+
+        if must_without_keys and not isinstance(value, UnknownFieldValue):
+            ok = ok and all([key not in data for key in must_without_keys])
+            msg = f'{", ".join(must_without_keys)} must not exist' if not ok else msg
+
+        if any_keys and isinstance(value, UnknownFieldValue):
```

Comment:
Why need to have `isinstance(value, UnknownFieldValue)` here for the `ANY` condition check?

Reply:
Hmm, indeed, seems that we don't need it.
Let' me try to remove it